### PR TITLE
Fix build with cmake under windows.

### PR DESCRIPTION
**Pull request by correa**
Fix build with MSVC, let cmake manage boost selection settings.

### CODE REVIEW

**Comment by correa** [comment]
This fixes issue #181 "Cant build libtorrent with Visual Studio 2015.Getting linking error."

File: CMakeLists.txt [hunk 1]
```diff
@@ -252,22 +256,19 @@ endif()
...
-	add_definitions(-DBOOST_SYSTEM_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_RANDOM_DYN_LINK)
-endif()
-
 if (WIN32)
-	target_link_libraries(torrent-rasterbar wsock32 ws2_32)
+	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi)
 	add_definitions(-D_WIN32_WINNT=0x0600)
 	# prevent winsock1 to be included
-	add_definitions(-DWIN32_LEAN_AND_MEAN)
+	add_definitions(-DWIN32_LEAN_AND_MEAN -DTORRENT_EXPORT_EXTRA)
```
Comment (arvidn) [inline CMakeLists.txt]: this define should only be added when building tests. is there a way to distinguish that? it's really not a good idea to include it for production builds, it exposes lots of internal functions and classes.

**Comment by correa** [comment]
Without TORRENT_EXPORT_EXTRA there is a link problem in cpuid.cpp mmx function detection in the user code. Am I doing something wrong?

**Comment by arvidn** [comment]
what's the error?

**Comment by arvidn** [comment]
the appveyor build of the examples, with shared linking, is supposed to catch issues like that. Since that's passing, it's possible there's a configuration issue in the cmake files causing it too

**Comment by correa** [comment]
I've updated the patch, thanks for noticing it. It was defined in user code as well, probably what caused the error. Removed from both and it builds fine.

**Pull request merged**